packages/staking/src/catalyst/translators/state.tsx

```diff
@@ -0,0 +1,78 @@
+import {freeze, produce} from 'immer'
+
+export const catalystReducer = (
+  state: CatalystState,
+  action: CatalystAction,
+) => {
+  return produce(state, (draft) => {
+    switch (action.type) {
+      case CatalystActionType.PinChanged:
+        draft.pin = action.pin
+        break
+
+      case CatalystActionType.VotingKeyEncryptedChanged:
+        draft.votingKeyEncrypted = action.votingKeyEncrypted
+        break
+
+      case CatalystActionType.Reset:
+        draft.pin = catalystDefaultState.pin
+        draft.votingKeyEncrypted = catalystDefaultState.votingKeyEncrypted
+        break
+
+      default:
+        throw new Error(`CatalystReducer invalid action`)
+    }
+  })
+}
+
+export const catalystDefaultState: Readonly<CatalystState> = freeze(
+  {
+    pin: null,
+    votingKeyEncrypted: null,
+  },
+  true,
+)
+
+export type CatalystState = {
+  pin: string | null
```

Comment:
Pin typed is an input, it should match with Pin generated, I'd say to create as a full input state, touched, error ....